framework/PostProcessors/FTstructure.py

```diff
@@ -0,0 +1,160 @@
+
+class FTstructure():
+  """
+    This is the base class of the FT structure which actually handles FT structures which is used by the FTimporter and the FTmodel
+  """
+  def __init__(self, inputs, topEventID):
+    """
+      This method executes the postprocessor action.
+      @ In,  inputs, list, list of file objects
+      @ Out, outcome, dict, dict containing the processed FT
+    """
+    self.basicEvents = []
+    self.houseEvents = {}
+    self.gateList    = {}
+    self.gateID      = []
+    self.topEventID  = topEventID
+
+    for file in inputs:
+      faultTree = ET.parse(file.getPath() + file.getFilename())
+      faultTree = findAllRecursive(faultTree,'opsa-mef')
+
+      for gate in findAllRecursive(faultTree[0], 'define-gate'):
+        FTGate = FTgate(gate)
+        self.gateList[gate.get('name')] = FTGate
+        self.gateID.append(gate.get('name'))
+
+      for basicEvent in findAllRecursive(faultTree[0], 'basic-event'):
+        self.basicEvents.append(basicEvent.get('name'))
+
+      for houseEvent in findAllRecursive(faultTree[0], 'define-house-event'):
+        value = houseEvent.find('constant').get('value')
+        if value in ['True','true']:
+          value = 1.
+        elif value in ['False','false']:
+          value = 0.
+        else:
+          raise IOError('FTImporterPostProcessor Post-Processor ' + self.name + ': house event ' + str(basicEvent.get('name')) + ' has a not boolean value (True or False)')
+        self.houseEvents[houseEvent.get('name')] = value
+
+    if not self.topEventID in self.gateID:
+      raise IOError('FTImporterPostProcessor: specified top event ' + str(self.topEventID) + ' is not contained in the fault-tree; available gates are: ' + str(self.gateID))
+
+  def returnDict(self):
+    """
+      This method returns
+      @ In,  None
+      @ Out, outcome, dict, dictionary containing
+    """
+    self.FTsolver()
+    outcome = self.constructData()
+    return outcome
+
+  def FTsolver(self):
+    """
+      This method determines the ordered sequence of gates to compute in order to solve the full FT.
+      The determined ordered sequence is stored in self.gateSequence.
+      @ In,  None
+      @ Out, None
+    """
+    self.gateSequence = []
+    availBasicEvents = copy.deepcopy(self.basicEvents)
+    availBasicEvents = availBasicEvents + self.houseEvents.keys()
+    counter = 0
+    while True:
+      complete=False
+      for gate in self.gateList.keys():
+        if set(self.gateList[gate].returnArguments()) <= set(availBasicEvents):
+          self.gateSequence.append(gate)
+          availBasicEvents.append(gate)
+        if set(availBasicEvents) == set(self.basicEvents+self.gateID+self.houseEvents.keys()):
+          complete=True
+          break
+        if counter > len(self.gateList.keys()):
+          raise IOError('FTImporterPostProcessor Post-Processor ' + self.name + ': the provided FT cannot be computed')
+        counter += 1
+      if complete:
```

Comment:
This is never reached...since you break above.... 
Actually setting complete to True is not usefult...

I think the complete variable is not needed (if you do not put it in the while loop)